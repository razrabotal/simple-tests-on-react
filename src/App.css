#q-simple {
  font-family: Verdana;
  max-width: 800px;
  box-sizing: border-box;
}

/* Reset styles */
#q-simple *,
#q-simple *:after,
#q-simple *:before {
  box-sizing: inherit;
}

#q-simple ul {
  padding: 0;
  margin: 0;
}
#q-simple li {
  list-style: none;
}
#q-simple p {
  margin: 0;
}


/* Main styles */
.q-main-title {
  margin-bottom: 40px;
}
.q-main-title-h1 {
  font-size: 40px;
  font-weight: 600;
  margin-bottom: 16px;
}
.q-main-title-p {
  font-size: 16px;
  line-height: 1.4;
}


.q-container {
  display: flex;
  flex-direction: column;
  margin-bottom: 40px;
}
.q-final {
  /* display: flex; */
  text-align: center;
  justify-content: space-between;
}


:root {
  --green: #00bf0b;
}

.q-result-text {
  font-size: 24px;
  margin-bottom: 16px!important;
}

.q-reload {
  padding: 8px 16px;
  box-sizing: content-box;
}
.refresh.icon {
  color: #444;
  position: relative;
  margin-left: 4px;
  margin-top: 3px;
  width: 16px;
  height: 16px;
  border-radius: 50%;
  border-top: solid 1px currentColor;
  border-bottom: solid 1px currentColor;
  border-left: solid 1px transparent;
  border-right: solid 1px currentColor;
  transition: 0.2s;
  box-sizing: content-box!important;
}
.refresh.icon:before {
  content: '';
  position: absolute;
  left: 1px;
  top: 12px;
  width: 6px;
  height: 5px;
  border-top: solid 1px currentColor;
  border-left: solid 1px currentColor;
  -webkit-transform: rotate(-12.5deg);
  transform: rotate(-12.5deg);
  box-sizing: content-box;
}
.q-reload:hover .refresh.icon{
  transform: rotate(15deg);
  color: var(--green);
  cursor: pointer;
}

.q-result-counts {
  font-size: 21px;
  letter-spacing: 2px;
  text-align: center;
  padding-top: 30%;
}
.q-result-percent {
  text-align: center;
  font-size: 14px;
}



.reload-icon {
  width: 24px;
  fill:none;
  stroke:#444;

  stroke-linejoin:round;
  stroke-miterlimit:10;
  stroke-width: 0.75;

  transform: rotate(0deg);
  transition: 0.2s;
}

.reload-icon path {
  /* animation: dash2 0.5s ease-in forwards; */
}

@keyframes dash2 {
  to {
    stroke-dashoffset: 0;
  }
}
 
.reload-icon polyline {
  /* animation: dash2 0.5s 0.5s ease-in forwards; */
}

.q-reload:hover .reload-icon {
  transform: rotate(15deg);
  stroke: var(--green);
  cursor: pointer;
}





.q-result-circle { 
  position: relative;
  width: 100px;
  height: 100px;
  margin: 0 auto 16px;
}

.svg-result {
  width: 100px;
  fill:none;
  stroke:var(--green);
  position: absolute;
  left: 0;
  stroke-dasharray: 30.25;
  stroke-dashoffset: 31;
  

}
.svg-result--background {
  position: absolute;
  z-index: -1;
  left:0;
  stroke:#eee;
  stroke-dashoffset: 0;
}

@keyframes circle1 { to { stroke-dashoffset:1 ; }}
@keyframes circle2 { to { stroke-dashoffset:2 ; }}
@keyframes circle3 { to { stroke-dashoffset:3 ; }}
@keyframes circle4 { to { stroke-dashoffset:4 ; }}
@keyframes circle5 { to { stroke-dashoffset:5 ; }}
@keyframes circle6 { to { stroke-dashoffset:6 ; }}
@keyframes circle7 { to { stroke-dashoffset:7 ; }}
@keyframes circle8 { to { stroke-dashoffset:8 ; }}
@keyframes circle9 { to { stroke-dashoffset:9 ; }}
@keyframes circle10 { to { stroke-dashoffset:10 ; }}
@keyframes circle11 { to { stroke-dashoffset:11 ; }}
@keyframes circle12 { to { stroke-dashoffset:12 ; }}
@keyframes circle13 { to { stroke-dashoffset:13 ; }}
@keyframes circle14 { to { stroke-dashoffset:14 ; }}
@keyframes circle15 { to { stroke-dashoffset:15 ; }}
@keyframes circle16 { to { stroke-dashoffset:16 ; }}
@keyframes circle17 { to { stroke-dashoffset:17 ; }}
@keyframes circle18 { to { stroke-dashoffset:18 ; }}
@keyframes circle19 { to { stroke-dashoffset:19 ; }}
@keyframes circle20 { to { stroke-dashoffset: 20; }}
@keyframes circle21 { to { stroke-dashoffset: 21; }}
@keyframes circle22 { to { stroke-dashoffset: 22; }}
@keyframes circle23 { to { stroke-dashoffset: 23; }}
@keyframes circle24 { to { stroke-dashoffset: 24; }}
@keyframes circle25 { to { stroke-dashoffset: 25; }}
@keyframes circle26 { to { stroke-dashoffset: 26; }}
@keyframes circle27 { to { stroke-dashoffset: 27; }}
@keyframes circle28 { to { stroke-dashoffset: 28; }}
@keyframes circle29 { to { stroke-dashoffset: 29; }}
@keyframes circle30 { to { stroke-dashoffset: 30; }}
@keyframes circle31 { to { stroke-dashoffset: 31; }}