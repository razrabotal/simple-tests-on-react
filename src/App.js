import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';

// class App extends Component {
//   render() {
//     return (
//       <div className="App">
//         <header className="App-header">
//           <img src={logo} className="App-logo" alt="logo" />
//           <h1 className="App-title">Welcome to React</h1>
//         </header>
//         <p className="App-intro">
//           To get started, edit <code>src/App.js</code> and save to reload.
//         </p>
//       </div>
//     );
//   }
// }

class App extends Component{
	constructor(props){
		super(props);
    this.currentIteration = this.currentIteration.bind(this)
    
		this.state = { 
      dataCounter: 1,
      lol : props.prueba.questions,
      results: props.prueba.results,
      current: 1,
      points: 0,     
      list: [],
      showList: []
    };
	} 
  
  componentWillMount() {
    // console.log(this.ref.innerHTML)
    // console.log(ReactDOM.findDOMNode(this).innerHTML)
  }
  componentDidMount() {
    
    console.log(this.props.prueba)
    // console.log(this.ref.innerHTML)
    
    
//     this.addToList(0); 
    
//     this.setState({
//       showList: [...this.state.showList, <Question next2={this.state.lol[0]} iteration={this.currentIteration}/>]
//     })
    
    this.setState({
      list: [...this.state.lol]
    })
  }
  
  reload() {
    this.setState({
      points: 0,
      current: 1,
      dataCounter: this.state.dataCounter + 1
    });
    
  }
  
  currentIteration(e, correct) {
    e.preventDefault();
   
    if(correct) {
       this.setState({
        points: this.state.points + 1
        })
      }

      this.setState({
        current: this.state.current + 1,
      })
  }
  
  addToList(id) {
    var element = this.state.lol[id];

    this.setState({
        list: [element, ...this.state.list]
    });
    
    console.log(this.state.list);
  }

   render() {
    // var showCurrent = this.state.lol.map((que)=>{
    //       if(this.state.current >= que.id) {
    //          return <Question next2={que} iteration={this.currentIteration}/>;
    //       }
    //       if(this.state.current === this.state.lol.length) {
    //         return <p>Все</p>
    //       }
    //     });
     
//     var qList = (() => {
//       if(this.state.current > this.state.lol.length) {
//         return <p>Все</p>
//       } else {
//         return showCurrent;
//       }  
//     })();
    
     // var showList = this.state.list.map((que)=>{
     //    return <Question next2={que} iteration={this.currentIteration}/>;
     // });
     
     var showList = this.state.list.map((que,index)=>{
          if(this.state.current >= que.id) {
             return <Question 
                key={`${this.state.dataCounter}_${index}`} 
                selected={false} 
                next2={que} 
                cur={this.state.current} 
                iteration={this.currentIteration}
                length={this.state.lol.length}
               />;
          }
          return null;
      });
     
     var showResultText = (() => {
       if(this.state.results) {
         for(var i of this.state.results) {
           if(i.points <= this.state.points) {
             return i.text;
           }
         }
       }
       return '';
     })();
     
     var showFinal = (this.state.current === this.state.lol.length+1) ? 
         <div className="q-result">
           <p>Вы набрали баллов: {this.state.points}</p>
           <p className="q-result-text">{showResultText}</p>
         </div> : '';
     
     var showProgress = (this.state.current > this.state.list.length) ? 
         <div className="q-reload" onClick={e => this.reload(e)}>
           <div class="refresh icon"></div>
         </div> : '';
     // <input type="button" className="q-reload" value="По новой" />;
     
     // var showList = this.state.showList;
     
     
    
    return (  
         <div>  
          
        <div className="q-main-title">
          <h1 className="q-main-title-h1">{this.props.prueba.title}</h1>
          <p className="q-main-title-p">{this.props.prueba.descr}</p>
         </div>
          <div className="q-container">
          { showList }
            <div className="q-final">
              { showFinal }
              <p>{ showProgress }</p>
            </div>    
          </div>
        
        
        
          
         </div>
    );
   }
}

class Question extends React.Component{
	constructor(props){
		super(props);
      
		this.state = { 
      que: props.next2,
      site: 'lol',
      comment: '',
      
      selectItem: '',
      selected: this.props.selected
    };
    
	}
  
  onChanged(e, comment, correct, id) { 
    
    if(this.state.selected === false) {
      this.setState({
        selected: true,
        selectItem: id,
        comment: comment
      });
      
      this.props.iteration(e, correct);
    }    
  }
 
   render() {
     

    
     var that = this;
    var que = this.state.que;
    // var iter = this.props.iteration;
     
    var qList = que.an.map(function(item,index){
   
      var qTrue = that.state.selected && item.correct ? 'q-true' : '',
      // qTrue = (that.state.selectItem === item.id && item.correct) ? 'q-true' : '',
          qSelect = (that.state.selected) ? 'q-select' : '',
          qUserSelect = that.state.selectItem === item.id ? 'q-user-select' : '';
      
      const classesButton = `q-button ${qSelect}`,
            classesLi = `q-list-item ${qUserSelect} ${qTrue}`;
    
          
        return <li key={index} className={classesLi}>
          <div       
            className={classesButton}
            onClick={e => that.onChanged(e, item.comment, item.correct, item.id)}
            >{item.text}</div>
        </li>; 
     })
    
    var imgRender = (que.img) ? <div className="q-img"><img alt="" src={que.img}/></div> : '';
      
    return (
      <div className="q-question">
           
        {imgRender}
        
        <div className="q-top">
          <div className="q-title">{this.state.que.q}</div>
          <div className="q-number">{this.state.que.id}/{this.props.length}</div>
        </div>
        
        
        <ul className="q-list">
          { qList }
        </ul>
        <p>{this.state.comment}</p>
      </div>
    );
   }
}





export default App;