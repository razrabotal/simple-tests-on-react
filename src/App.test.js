import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
// import Title from './App';
import parseInner from './parser';


import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { shallow, render, mount } from 'enzyme';
global.shallow = shallow;
global.render = render;
global.mount = mount; 

it('should render a title', () => { 
  const wrapper = shallow(
    <App/>
  ); 
  expect(wrapper).toMatchSnapshot();
});

it('renders without crashing', () => {
  var div = document.createElement('div');
  div.id = "root";
  ReactDOM.render(<App prueba={parseInner(`
  ## Приветствие
  # Тест на знание приветствия
   
     Привет
     http://hronika.info/uploads/posts/2016-01/1452884960_holms2.jpg
     - Привет - не
     - Привки - тира уркиау
     + Хай - Да, правильно
     - Хаюшки - ноу
  
   Как дела?
     - Хорошо
     + не хорошо
     - плохо
 
 2 - Все ок
 1 - Ну норм
 0 - Отстой
  `)} />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders without crashing', () => {
  var div = document.createElement('div');
  div.id = "root";
  ReactDOM.render(<App data={parseInner(`
  ## Приветствие
  # Тест на знание приветствия
   
     Привет
     http://hronika.info/uploads/posts/2016-01/1452884960_holms2.jpg
     - Привет - не
     - Привки - тира уркиау
     + Хай - Да, правильно
     - Хаюшки - ноу
 
   Как дела?
     - Хорошо
     + не хорошо
     - плохо
 
 2 - Все ок
 1 - Ну норм
 0 - Отстой
  `)} />, div);
  ReactDOM.unmountComponentAtNode(div);
});