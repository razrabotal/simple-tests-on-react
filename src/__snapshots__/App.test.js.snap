// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Title />,
        <Circles />,
        <div
          className="q-container"
>
          <Final />
          <div
                    className="q-question-list"
          >
                    
          </div>
</div>,
        <Next />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Final />,
            <div
              className="q-question-list"
>
              
</div>,
          ],
          "className": "q-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "q-question-list",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Title />,
          <Circles />,
          <div
            className="q-container"
>
            <Final />
            <div
                        className="q-question-list"
            >
                        
            </div>
</div>,
          <Next />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Final />,
              <div
                className="q-question-list"
>
                
</div>,
            ],
            "className": "q-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "q-question-list",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
